improved options dialog with searching options

HUSSAR-mtrela opened the pull request:
- improved scrolling performance in gird control
- improved default padding in tooltips
- ignoring DEV files in root of the repo
- fixed UI layout/clipping issues in options dialog
- replaced tabs with collapsible list of options (see screenshot 1)
- add missing tooltips for a few options
- add searching/filtering of all options by their names or values (see screenshot 2 and 3)
- add possibility of creating new group of options, e.g. mods can hook the [lua/options/options.lua](https://github.com/FAForever/fa/blob/deploy/fafdevelop/lua/options/options.lua#L167) file and add new group of options inside of the [options](https://github.com/FAForever/fa/blob/deploy/fafdevelop/lua/options/options.lua#L167) table such as [my_mod_options](https://github.com/FAForever/fa/blob/hussar-game-options/lua/ui/dialogs/options.lua#L262) (see screenshot 4)

Screenshot 1 - the new options dialog with collapsible list of options:
![image](https://github.com/FAForever/fa/assets/11665978/21222431-1e52-4828-95da-2b21c6894787)

Screenshot 2 - the new options dialog with searching by option's name:
![image](https://github.com/FAForever/fa/assets/11665978/14333713-e136-4c60-b319-6cd00d7c15f5)

Screenshot 3 - the new options dialog with searching by option's value:
![image](https://github.com/FAForever/fa/assets/11665978/76982016-425d-46de-b775-eb6637b04edb)

Screenshot 4 - the new options dialog with custom mod options:
![image](https://github.com/FAForever/fa/assets/11665978/f2ff9881-c699-4ad5-ad3e-288db1cbf03f)

Screenshot 5 - the old options dialog:
![image](https://github.com/FAForever/fa/assets/11665978/3d9ed17f-c735-4dae-8b82-1d6221f80fe7)

Garanas commented on lua/maui/grid.lua:
```diff
@@ -252,19 +252,26 @@ Grid = ClassUI(Group) {
                         local vertPad = math.max(0, (self._itemHeight - control.Height()) / 2)
-                        control.Left:Set(function() return math.floor(((column - self._top["Horz"]) * self._itemWidth) + self.Left() + horzPad) end)
-                        control.Top:Set(function() return math.floor(((rowumn - self._top["Vert"]) * self._itemHeight) + self.Top() + vertPad) end)
+                        control.Left:Set(function() return math.floor(((column - scrollHorz) * self._itemWidth) + self.Left() + horzPad) end)
```
These closures are likely the most expensive part of this function, not that you have to fix that but just pointing it out

Garanas reviewed (commented):
Looking great! It appears we're not 'properly' using the background texture, and instead apply black squares on top of it.

![image](https://github.com/FAForever/fa/assets/15778155/0952eada-dd7a-4463-a258-e135d6c23641)

Would it be possible to properly use the background texture, put the search at the top and shorten the list with 2 or 3 elements?

Garanas commented:
Looking at the current options menu:

![image](https://github.com/FAForever/fa/assets/15778155/f5ee9102-0a23-453f-8bb6-e74646ccbaba)

@HUSSAR-mtrela would it be possible to use the texture space accordingly by having the search + game options in the background area that is designed to hold the options? I don't know by heart what the dialog texture looks like exactly, I'll try to find it

HUSSAR-mtrela commented:
@Garanas the background texture has semi-transparent region where tab controls used to be and the same semi-transparent region above Reset/OK/Cancel/Apply buttons (see 2nd screenshot below) that is why I added that black bitmap to fill those regions. Otherwise search box would not be very visible and there would be a lot of wasted space that we can now use for displaying options. I have just fixed the alignment of the black fill with the background texture (see my screenshot below)

Latest Option Dialog with fixed alignment of fill bitmap:
![image](https://github.com/FAForever/fa/assets/11665978/1a805c68-78e4-4f03-b3b5-e8d0daf8eeb1)

Original Texture without tabs and options list:
![image](https://github.com/FAForever/fa/assets/11665978/efe6c9ee-4bbc-4347-a573-825ee09e9a26)

By they way, I added an option to open this dialog via key binding
![image](https://github.com/FAForever/fa/assets/11665978/d565d475-f65a-4b85-a155-ac45700b4ab1)

HUSSAR-mtrela commented:
@Garanas If you like I can replace the styling of the Option window (which does not changes based on selected faction) with styling of the Keybinding window which matches currently selected faction.

BlackYps commented:
I have to say, visually it looks worse than what we had before. Was there a reason to replace the scroll bar and the borders around the individual options?
It also looks like you tried to squeeze every inch of vertical space out of it. The black area now almost touches the buttons, which doesn't look good

MrRowey commented:
@HUSSAR-mtrela this can be updated to use the tab option as I agree with Blackyps that it dose not look as visually good.

MrRowey commented:
Closing for now Can be reopened if updated with the if updated with the feedback
